File: tea-system/internal/models/custom_product.go
package models

import (
	"time"
)

// CustomProductStatus
const (
	CustomProductStatusDraft     = "draft"
	CustomProductStatusPublished = "published"
	CustomProductStatusArchived  = "archived"
)

// CustomProduct — 定制报价商品表（完整 26+ 字段，含溯源）
type CustomProduct struct {
	ID                  uint64    `gorm:"primaryKey;column:id" json:"id"`
	ProductToken        *string   `gorm:"column:product_token;uniqueIndex;size:64" json:"product_token,omitempty"`
	Version             int       `gorm:"column:version;default:1" json:"version"`
	IsBespoke           bool      `gorm:"column:is_bespoke;default:true" json:"is_bespoke"`
	NonRefundable       bool      `gorm:"column:non_refundable;default:true" json:"non_refundable"`
	Title               string    `gorm:"column:title;not null;size:200" json:"title"`
	RawTeaSource        string    `gorm:"column:raw_tea_source;not null;type:text" json:"raw_tea_source"`
	CustomRequirement   string    `gorm:"column:custom_requirement;not null;type:text" json:"custom_requirement"`
	TeaType             string    `gorm:"column:tea_type;not null;size:20" json:"tea_type"` // raw_puer / ripe_puer / ancient_tree / vintage
	TeaShape            string    `gorm:"column:tea_shape;not null;size:20" json:"tea_shape"` // loose / cake / brick / tuo
	TeaShapeWeight      *int      `gorm:"column:tea_shape_weight" json:"tea_shape_weight,omitempty"`
	SmokedWithFlower    bool      `gorm:"column:smoked_with_flower;default:false" json:"smoked_with_flower"`
	FlowerType          string    `gorm:"column:flower_type;size:30" json:"flower_type,omitempty"` // jasmine / osmanthus / orchid / custom
	InnerPackaging      string    `gorm:"column:inner_packaging;not null;size:100" json:"inner_packaging"`
	OuterPackaging      string    `gorm:"column:outer_packaging;not null;size:100" json:"outer_packaging"`
	ProductCardText     string    `gorm:"column:product_card_text;type:text" json:"product_card_text,omitempty"`
	ProductCardFormat   string    `gorm:"column:product_card_format;size:20;default:'vertical'" json:"product_card_format"` // vertical / horizontal
	QrCodePosition      string    `gorm:"column:qr_code_position;not null;size:20" json:"qr_code_position"` // outer_front / outer_back / inner_front / hidden
	SKU                 string    `gorm:"column:sku;uniqueIndex;size:50" json:"sku,omitempty"`
	UnitPrice           float64   `gorm:"column:unit_price;not null;type:numeric(10,2)" json:"unit_price"`
	Quantity            int       `gorm:"column:quantity;not null" json:"quantity"`
	ShippingCost        float64   `gorm:"column:shipping_cost;not null;type:numeric(10,2)" json:"shipping_cost"`
	TotalAmount         float64   `gorm:"column:total_amount;not null;type:numeric(10,2)" json:"total_amount"`
	LeadTime            string    `gorm:"column:lead_time;not null;size:100" json:"lead_time"`
	Status              string    `gorm:"column:status;not null;size:20" json:"status"` // draft / published / archived
	CreatedByStaffID    *uint64   `gorm:"column:created_by_staff_id" json:"created_by_staff_id,omitempty"`
	ReviewedByStaffID   *uint64   `gorm:"column:reviewed_by_staff_id" json:"reviewed_by_staff_id,omitempty"`
	ReviewedAt          *time.Time `gorm:"column:reviewed_at" json:"reviewed_at,omitempty"`

	// 溯源字段
	HarvestDate     time.Time `gorm:"column:harvest_date;not null;type:date" json:"harvest_date"`
	RoastingDate    time.Time `gorm:"column:roasting_date;not null;type:date" json:"roasting_date"`
	MountainLocation string   `gorm:"column:mountain_location;not null;size:200" json:"mountain_location"`
	MasterName      string    `gorm:"column:master_name;not null;size:100" json:"master_name"`
	StorageLocation string    `gorm:"column:storage_location;not null;size:200" json:"storage_location"`
	SgsReportID     *uint64   `gorm:"column:sgs_report_id" json:"sgs_report_id,omitempty"`

	CreatedAt time.Time `gorm:"column:created_at;not null" json:"created_at"`
	UpdatedAt time.Time `gorm:"column:updated_at;not null" json:"updated_at"`
	DeletedAt *time.Time `gorm:"column:deleted_at" json:"-"`

	CreatedByStaff *Staff     `gorm:"foreignKey:CreatedByStaffID" json:"created_by_staff,omitempty"`
	ReviewedByStaff *Staff    `gorm:"foreignKey:ReviewedByStaffID" json:"reviewed_by_staff,omitempty"`
	SgsReport      *SgsReport `gorm:"foreignKey:SgsReportID" json:"sgs_report,omitempty"`
}

func (CustomProduct) TableName() string {
	return "custom_products"
}

File: tea-app-android/lib/models/custom_product.dart
import 'package:json_annotation/json_annotation.dart';

/// 100% 对齐 Go struct custom_products — 26 字段用户输入 + 自动生成字段
@JsonSerializable()
class CustomProduct {
  final int? id;
  final String? product_token;      // 自动生成 64 位
  final int? version;               // 自动版本号
  final bool? is_bespoke;
  final bool? non_refundable;

  // === 用户输入 25 字段 ===
  final String title;                           // 1
  @JsonKey(name: 'raw_tea_source') final String rawTeaSource;  // 2
  @JsonKey(name: 'custom_requirement') final String customRequirement; // 3
  @JsonKey(name: 'tea_type') final String teaType;             // 4: raw_puer / ripe_puer / ancient_tree / vintage
  @JsonKey(name: 'tea_shape') final String teaShape;           // 5: loose / cake / brick / tuo
  @JsonKey(name: 'tea_shape_weight') final int teaShapeWeight; // 6
  @JsonKey(name: 'smoked_with_flower') final bool smokedWithFlower; // 7
  @JsonKey(name: 'flower_type') final String? flowerType;      // 8
  @JsonKey(name: 'inner_packaging') final String innerPackaging;  // 9
  @JsonKey(name: 'outer_packaging') final String outerPackaging; // 10
  @JsonKey(name: 'product_card_text') final String? productCardText; // 11
  @JsonKey(name: 'product_card_format') final String productCardFormat; // 12
  @JsonKey(name: 'qr_code_position') final String qrCodePosition; // 13
  final double unitPrice;                                      // 14
  final int quantity;                                          // 15
  final double shippingCost;                                   // 16
  final String leadTime;                                       // 17
  @JsonKey(name: 'harvest_date') final String harvestDate;     // 18
  @JsonKey(name: 'roasting_date') final String roastingDate;   // 19
  @JsonKey(name: 'tea_garden_location') final String teaGardenLocation; // 20
  @JsonKey(name: 'master_name') final String masterName;       // 21
  @JsonKey(name: 'storage_location') final String storageLocation; // 22
  @JsonKey(name: 'sgs_report_id') final int? sgsReportId;      // 23
  final bool includeCustomLive;                                // 24
  final String? liveScheduledDate;                             // 25

  final double? totalAmount;
  final String? status;  // draft / published / archived
  final String? sku;     // auto-gen

  CustomProduct({
    this.id, this.product_token, this.version = 1, this.is_bespoke = true, this.non_refundable = true,
    required this.title, required this.rawTeaSource, required this.customRequirement,
    required this.teaType, required this.teaShape, this.teaShapeWeight = 357,
    this.smokedWithFlower = false, this.flowerType,
    required this.innerPackaging, required this.outerPackaging,
    this.productCardText, this.productCardFormat = 'vertical',
    this.qrCodePosition = 'outer_back',
    required this.unitPrice, required this.quantity, this.shippingCost = 0,
    required this.leadTime,
    required this.harvestDate, required this.roastingDate,
    required this.teaGardenLocation, required this.masterName, required this.storageLocation,
    this.sgsReportId, this.includeCustomLive = false, this.liveScheduledDate,
    this.totalAmount, this.status, this.sku,
  });

  Map<String, dynamic> toJson() => {
    'title': title, 'raw_tea_source': rawTeaSource, 'custom_requirement': customRequirement,
    'tea_type': teaType, 'tea_shape': teaShape, 'tea_shape_weight': teaShapeWeight,
    'smoked_with_flower': smokedWithFlower, 'flower_type': flowerType,
    'inner_packaging': innerPackaging, 'outer_packaging': outerPackaging,
    'product_card_text': productCardText, 'product_card_format': productCardFormat,
    'qr_code_position': qrCodePosition,
    'unit_price': unitPrice, 'quantity': quantity, 'shipping_cost': shippingCost,
    'lead_time': leadTime,
    'harvest_date': harvestDate, 'roasting_date': roastingDate,
    'tea_garden_location': teaGardenLocation, 'master_name': masterName,
    'storage_location': storageLocation, 'sgs_report_id': sgsReportId,
    'include_custom_live': includeCustomLive, 'live_scheduled_date': liveScheduledDate,
  };
}
